homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,154 @@
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        """Handle the initial step to select the location."""
+
+        if user_input:
+            self._latitude = user_input[CONF_LOCATION][CONF_LATITUDE]
+            self._longitude = user_input[CONF_LOCATION][CONF_LONGITUDE]
+            self._radius = user_input[CONF_LOCATION][CONF_RADIUS]
+
+            client: OpenAPI = OpenAPI()
+            self._stations = await client.get_devices_by_location(
+                self._latitude,
+                self._longitude,
+                radius=DistanceConverter.convert(
+                    self._radius,
+                    UnitOfLength.METERS,
+                    UnitOfLength.MILES,
+                ),
+            )
+
+            # Filter out indoor stations
+            self._stations = list(
+                filter(
+                    lambda station: not station.get(API_STATION_INFO, {}).get(
+                        API_STATION_INDOOR, False
+                    ),
+                    self._stations,
+                )
+            )
+
+            if self._stations:
+                return await self.async_step_station()
+
+            errors = {"base": "no_stations_found"}
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONF_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONF_LOCATION: {
+                    CONF_LATITUDE: self.hass.config.latitude,
+                    CONF_LONGITUDE: self.hass.config.longitude,
+                    CONF_RADIUS: CONF_RADIUS_DEFAULT,
+                }
+                if not errors
+                else {
+                    CONF_LATITUDE: self._latitude,
+                    CONF_LONGITUDE: self._longitude,
+                    CONF_RADIUS: self._radius,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONF_USER, data_schema=schema, errors=errors if errors else {}
+        )
+
+    async def async_step_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the second step to select the station."""
+
+        if user_input:
+            mac_address, station_name = user_input[CONF_STATION].split(",")
+            await self.async_set_unique_id(mac_address)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=station_name,
+                data={
+                    CONF_MAC: mac_address,
+                },
+            )
+
+        options: list[SelectOptionDict] = [
+            SelectOptionDict(
+                label=get_station_name(station),
+                value=f"{station[API_STATION_MAC_ADDRESS]},{get_station_name(station)}",
```

Comment:
Why is the mac address not enough as key. If it's only to get the name back when they user has selected a station, I think it would be better to find the station name in self._stations based on the selected MAC address.

Reply:
Because we need the station name for the title of the config entry. See [here](https://github.com/home-assistant/core/pull/105779/files#diff-3c9f07c74b4447e1eff413eeae362620fe4b8e29d2a5ed41b1fb385837cc55eeR126)

Reply:
He means to do a lookup for the name when creating the entry instead of having a whole options dict here with a value which is a combo of mac + name

Reply:
Why would I do another expensive network call when I create the entry when the information is already there?

Reply:
Fixed.